### PR TITLE
vc1: support multi slice decoding

### PR DESCRIPTION
**Pull request by YuJiankang**
patch to support vc1 multi slice, with this patch, the following file should be fixed:
SA00049, SA00058, SA00059, SA10097, SA10131
Signed-off-by: jkyu jiankang.yu@intel.com

### CODE REVIEW

File: codecparsers/vc1Parser.cpp [hunk 1]
```diff
@@ -748,6 +782,16 @@ namespace VC1 {
             }
         }
 
+        if (m_frameHdr.picture_type == FRAME_B) {
+            uint16_t bfraction;
+            if (!decodeVLCTable(br, &bfraction, BFractionVLCTable, 23 ))
```
Comment (xuguangxin) [inline codecparsers/vc1Parser.cpp]: 23 is magic number, could you use N_ELEMENT(BFractionVLCTable)

File: codecparsers/vc1Parser.cpp [hunk 2]
```diff
@@ -911,6 +956,21 @@ namespace VC1 {
...
         if (m_seqHdr.finterpflag)
             READ(m_frameHdr.interpfrm);
+        if ((m_frameHdr.fcm != FIELD_INTERLACE
+             && m_frameHdr.picture_type == FRAME_B)
+             || (m_frameHdr.fcm == FIELD_INTERLACE
+             && ((m_frameHdr.picture_type == FRAME_B)
+             || (m_frameHdr.picture_type == FRAME_BI)))) {
+
+            uint16_t bfraction;
+            if (!decodeVLCTable(br, &bfraction, BFractionVLCTable, 23 ))
```
Comment (xuguangxin) [inline codecparsers/vc1Parser.cpp]: magic number

File: codecparsers/vc1Parser.cpp [hunk 3]
```diff
@@ -748,6 +782,16 @@ namespace VC1 {
             }
         }
 
+        if (m_frameHdr.picture_type == FRAME_B) {
```
Comment (xuguangxin) [inline codecparsers/vc1Parser.cpp:799]: this is duplicate with L965 is it ok to wrapper it as function, such as decodeBFraction?

**Comment by YuJiankang** [comment]
patch is updated according to the comments, thanks

**Comment by xuguangxin** [comment]
+1

**Comment by xuguangxin** [comment]
merged as 3a8afcf171d3478cf1f1d4222db5f896d80ac866

**Pull request closed without merging**